apps/src/schoolInfo/hooks/useSchoolInfo.tsx

```diff
@@ -140,22 +194,26 @@ export function useSchoolInfo(initialState: SchoolInfoInitialState) {
 
   // Handle schoolId changes
   useEffect(() => {
-    sessionStorage.setItem(SCHOOL_ID_SESSION_KEY, schoolId);
-    if (!mounted.current) return;
-
+    handleSessionStorage(SCHOOL_ID_SESSION_KEY, schoolId);
     if (schoolId === NonSchoolOptions.NO_SCHOOL_SETTING) {
+      setSchoolName('');
       analyticsReporter.sendEvent(
         EVENTS.DO_NOT_TEACH_AT_SCHOOL_CLICKED,
         {},
         PLATFORMS.BOTH
       );
     } else if (schoolId === NonSchoolOptions.CLICK_TO_ADD) {
+      setSchoolName('');
       analyticsReporter.sendEvent(
         EVENTS.ADD_MANUALLY_CLICKED,
         {},
         PLATFORMS.BOTH
       );
     } else {
+      const name = schoolsList.find(school => school.value === schoolId)?.text;
+      if (name) {
+        setSchoolName(name);
+      }
```

Comment:
I moved this logic to the useSchoolInfo hook so it's easier to test all together